### PR TITLE
Migrate to yarn v3

### PR DESCRIPTION
**Pull request by TomAFrench**
We're currently running on Yarn v1 which is now deprecated, this PR migrates to the latest version (3.2.0).

Yarn v2+ has many advantages over v1 including more flexible support for monorepos so we can perform easier testing/linting/etc than currently.

### CODE REVIEW

**Comment by TomAFrench** [comment]
I've broken [775e24d](https://github.com/actualbudget/actual/pull/50/commits/775e24d0995c65f64e2bda16cd6f362ff312a383) through [3e41fa9](https://github.com/actualbudget/actual/pull/50/commits/3e41fa9b3e3b2415ddf0509a96505f236fa430d5) out into separate PRs (#53 and #55) as the issue I was fixing here also exists in master so we can fix it without having to jump a couple of major versions in yarn at the same time.

**Comment by gbdavidx** [comment]
will v3 add the capability to connect to bankaccounts via plaid to sync automatically?

**Comment by TomAFrench** [comment]
No this is purely a "behind the scenes" change and won't have any effect on the app.

**Comment by TomAFrench** [comment]
@jlongster I'd recommend prioritising this PR to be merged. Having stricter (more predictable) dependency resolution would likely help with fixing the issues in #108 and keeping this up to date is quite painful due to the change in lockfile format.

File: bin/package [hunk 1]
```diff
@@ -80,27 +80,14 @@ if [[ $CI != true && "$OSTYPE" == "darwin"* ]]; then
...
-(
-  cd packages/loot-core;
-  NODE_ENV=production yarn build:node
-)
+yarn workspace @actual-app/web build
 
-(
-  cd packages/desktop-client;
-  yarn build
-)
```
Comment (jlongster) [inline bin/package:103]: Why'd you delete this?

File: bin/package-browser [hunk 2]
```diff
@@ -56,19 +56,8 @@ if [ $CI != true ]; then
     yarn lint
 fi
 
-(
-  cd packages/loot-design;
-  ../../node_modules/.bin/patch-package
-)
```
Comment (jlongster) [inline bin/package-browser:62]: Looks like `patch-package` isn't running anymore

**Review by jlongster — COMMENTED** [review]
This looks great; let's work on getting this in. (sorry for the hassle, this is a PR that is annoying to keep alive)

I want to read up on yarn 3 again to refresh my memory. I was intentionally staying away from yarn 2 because I really didn't like how it worked. Maybe eventually we do move back to npm. It sounds like yarn 3 is more sane, and `nodeLinker: node-modules` allows us to have a similar strategy as yarn 1 correct? I don't really want to move to PnP

Does anything fundamental change for us here? My guess is no -- what are the main differences though?

File: packages/import-ynab4/importer.js [hunk 3]
```diff
@@ -2,8 +2,8 @@ const d = require('date-fns');
 const normalizePathSep = require('slash');
 const uuid = require('uuid');
 const AdmZip = require('adm-zip');
-const actual = require('@actual-app/api/methods');
-const { amountToInteger } = require('@actual-app/api/utils');
+const actual = require('@actual-app/api');
+const amountToInteger = actual.utils.amountToInteger;
```
Comment (jlongster) [inline packages/import-ynab4/importer.js:6]: Why this change?

Comment (TomAFrench) [reply packages/import-ynab4/importer.js:6]: I get the below error with the original code.

```
ERROR in ./packages/import-ynab4/importer.js
Module not found: Error: Can't resolve '@actual-app/api/methods' in '/home/tom/Programming/actual/packages/import-ynab4'
 @ ./packages/import-ynab4/importer.js 21:15-49
 @ ./packages/loot-core/src/server/main.js
```

I imagine that it's yarn starting enforcing that you go in through the main entrypoint but I didn't dig too much into it.

File: bin/package-browser (hunk 2 above)
Comment (TomAFrench) [reply bin/package-browser:62]: The downshift patch was the only one which was being run in the `loot-design` context but that's been moved up to the monorepo root so this is unneeded now.

Comment (TomAFrench) [reply bin/package-browser:62]: I moved the patch file up as the dependency was now stored in the monorepo root's version of `node_modules` and so the patch moved to match.

File: bin/package (hunk 1 above)
Comment (TomAFrench) [reply bin/package:103]: These 4 lines are equivalent to running `yarn workspace @actual-app/web build` in the monorepo root. I felt that this script was more readable using this format rather than cding into each package.

**Comment by TomAFrench** [comment]
> I want to read up on yarn 3 again to refresh my memory. I was intentionally staying away from yarn 2 because I really didn't like how it worked. Maybe eventually we do move back to npm. It sounds like yarn 3 is more sane, and `nodeLinker: node-modules` allows us to have a similar strategy as yarn 1 correct? I don't really want to move to PnP

Yeah, I think they've put a fair bit of work in getting v3 to be a smaller jump from v1 than v2 was.

> Does anything fundamental change for us here? My guess is no -- what are the main differences though?

There aren't any fundamental changes which aren't opt-in afaik. The closest thing to this is dependency resolution is stricter. Iit seems like with yarn v1 if one workspace has a dependency then all the other workspaces can make use of it, in yarn v3 we need to specify that dependency in each workspace.

The other main benefit is better monorepo support so we can run tasks across multiple workspaces without having to enumerate them.

**Comment by jlongster** [comment]
I've been trying out yarn v3 a bunch and I feel comfortable merging this in. Thanks for all the work!!
